src/imap.rs

```diff
@@ -375,6 +394,8 @@ impl Imap {
         let imap_user: &str = config.lp.user.as_ref();
         let imap_pw: &str = config.lp.password.as_ref();
 
+        self.login_ratelimit.write().await.send();
```

Comment:
Maybe still call it `connect_ratelimit` and move right after `let client = connection_res?;`? Otherwise if some additional checks will be inserted in the code before this, a busy connection loop is still possible